mne/utils/tests/test_config.py

```diff
@@ -87,3 +87,18 @@ def test_sys_info():
 
     if platform.system() == 'Darwin':
         assert 'Platform:      macOS-' in out
+
+
+def test_get_subjects_dir(monkeypatch):
+    """Test get_subjects_dir()."""
+    # String
+    subjects_dir = '/foo'
+    assert get_subjects_dir(subjects_dir) == subjects_dir
+
+    # Path
+    subjects_dir = Path('/foo')
+    assert get_subjects_dir(subjects_dir) == str(subjects_dir)
+
+    # `None`
+    monkeypatch.delenv('SUBJECTS_DIR', raising=False)
+    assert get_subjects_dir() is None
```

Comment:
Reverting the `tmpdir` was an error, anyone with a `.mne/mne-python.json` file with `SUBJECTS_DIR` will have this test fail, such as for me locally on my macOS machine where I define it to be the MNE-sample-data dir:
```
____________________________________________________________________ test_get_subjects_dir _____________________________________________________________________
mne/utils/tests/test_config.py:104: in test_get_subjects_dir
    assert get_subjects_dir() is None
E   AssertionError: assert '/Users/larsoner/mne_data/MNE-sample-data/subjects' is None
E    +  where '/Users/larsoner/mne_data/MNE-sample-data/subjects' = get_subjects_dir()
```
I would just `monkeypatch.setenv('_MNE_FAKE_HOME_DIR', str(tmpdir))` so that `get_config` reroutes to a fake home dir

Reply:
Aaarghhh! Will fix in a followup PR